### PR TITLE
[BEAM-4176] Tests for running Python on Flink.

### PR DESCRIPTION
**Pull request by angoenka**
This provides a fairly comprehensive tests of the basic runner compatibility expectations for Python on Flink. Surprisingly, the suite only takes a minute or so to run, so totally feasible to run on presubmit.

Copy of https://github.com/apache/beam/pull/5942
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by angoenka** [comment]
R: @tweise @robertwb @bsidhom

File: sdks/python/apache_beam/runners/portability/flink_runner_test.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  class FlinkRunnerTest(portable_runner_test.PortableRunnerTest):
+    _use_grpc = True
+    _use_subprocesses = True
+
+    @classmethod
+    def _subprocess_command(cls, port):
+      return [
+          'java',
+          '-jar', flinkJobServerJar,
+          '--artifacts-dir', '/tmp/flink',
```
Comment (tweise) [inline sdks/python/apache_beam/runners/portability/flink_runner_test.py]: should probably use the system tmp dir here?

File: sdks/python/apache_beam/runners/portability/flink_runner_test.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          '--job-host', 'localhost:%s' % port,
+      ]
+
+    @classmethod
+    def get_runner(cls):
+      return portable_runner.PortableRunner()
+
+    def create_options(self):
+      options = super(FlinkRunnerTest, self).create_options()
```
Comment (tweise) [inline sdks/python/apache_beam/runners/portability/flink_runner_test.py]: We should support the `streaming` option here and cover the streaming mode.

File: sdks/python/apache_beam/runners/portability/flink_runner_test.py (hunk 1 above)
Comment (angoenka) [reply sdks/python/apache_beam/runners/portability/flink_runner_test.py]: Makes sense.

File: sdks/python/apache_beam/runners/portability/flink_runner_test.py (hunk 2 above)
Comment (angoenka) [reply sdks/python/apache_beam/runners/portability/flink_runner_test.py]: Done

**Pull request merged**